drift-codec/src/main/java/io/airlift/drift/codec/internal/compiler/ThriftCodecByteCodeGenerator.java

```diff
@@ -1038,8 +1044,10 @@ private static boolean needsCodec(ThriftFieldMetadata fieldMetadata)
 
     private static ParameterizedType toCodecType(ThriftStructMetadata metadata)
     {
-        String className = type(metadata.getStructClass()).getClassName();
-        return typeFromPathName(PACKAGE + "/" + className + "Codec");
+        String className = type(metadata.getStructClass()).getClassName()
+                + "_" + Instant.now().atZone(UTC).format(TIMESTAMP_FORMAT)
+                + "_" + CLASS_ID.incrementAndGet();
+        return typeFromPathName(PACKAGE + "/" + className + "Codec" + metadata.getStructType().hashCode());
```

Comment:
Using a sanitized version of the struct type name seems better than a hash code. Something like this (untested):
```java
metadata.getStructType().getTypeName().replaceAll("[^a-zA-Z0-9]+", "_");